src/algebra/order/monoid_lemmas_zero_lt.lean

```diff
@@ -664,8 +671,35 @@ lemma left.mul_lt_one_of_lt_of_lt [pos_mul_strict_mono α]
   (ha : a < 1) (hb : b < 1) (a0 : 0 < a) : a * b < 1 :=
 mul_lt_of_lt_of_lt_one ha hb a0
 
-/-! Lemmas of the form `b ≤ c → 1 ≤ a → 0 < c → b ≤ c * a`,
-which assume left covariance. -/
+-- proven with `a0 : 0 ≤ a` and `c0 : 0 ≤ c` as `mul_le_of_le_of_le_one'`
+lemma preorder.mul_le_of_le_of_le_one' [pos_mul_mono α] [mul_pos_mono α]
+  (bc : b ≤ c) (ha : a ≤ 1) (a0 : 0 < a) (c0 : 0 < c) : b * a ≤ c :=
```

Comment:
I'm confused about why we would ever want this if we have a version with 0<=a and 0<=c, which is then strictly more applicable. Note that if `h : 0 < a` then `h.le` should be a proof of `0 \le a`. If you put this lemma later you could just prove it like this:
```lean
lemma preorder.mul_le_of_le_of_le_one' [pos_mul_mono α] [mul_pos_mono α]
  (bc : b ≤ c) (ha : a ≤ 1) (a0 : 0 < a) (c0 : 0 < c) : b * a ≤ c :=
mul_le_of_le_of_le_one' bc ha a0.le c0.le
```

So why do we need it?

Reply:
We have these lemmas (in `preorder` namespace) for `preorder` and some stronger lemmas like `mul_le_of_le_of_le_one'` for `partial_order`. See also docstring of this file and #14060.